ansible/roles/dovecot/tasks/main.yml

```diff
@@ -61,105 +69,79 @@
 - name: Update Ansible facts if they were modified
   meta: 'flush_handlers'
 
-- name: Divert original Dovecot configuration
+- name: Divert original Dovecot main configuration file
   command: dpkg-divert --quiet --local
-           --divert /etc/dovecot/conf.d/{{ item }}.dpkg-divert
-           --rename /etc/dovecot/conf.d/{{ item }}
-           creates=/etc/dovecot/conf.d/{{ item }}.dpkg-divert
-  with_flattened:
-    - [ '10-auth.conf', '10-mail.conf', '10-master.conf', '10-ssl.conf', '15-lda.conf' ]
-    - [ '{{ "20-imap.conf" if "imap" in dovecot_protocols else [] }}' ]
-    - [ '{{ "20-pop3.conf" if "pop3" in dovecot_protocols else [] }}' ]
-    - [ '{{ "20-lmtp.conf" if "lmtp" in dovecot_protocols else [] }}' ]
-    - [ '{{ "auth-deny.conf.ext" if "deny" in dovecot_user_accounts else [] }}' ]
-    - [ '{{ "auth-system.conf.ext" if "system" in dovecot_user_accounts else [] }}' ]
-    - [ '{{ "auth-sql.conf.ext" if "sql" in dovecot_user_accounts else [] }}' ]
-    - [ '{{ "auth-ldap.conf.ext" if "ldap" in dovecot_user_accounts else [] }}' ]
-    - [ '{{ "auth-checkpassword.conf.ext" if "checkpassword" in dovecot_user_accounts else [] }}' ]
-    - [ '{{ [ "20-managesieve.conf", "90-sieve.conf", "90-sieve-extprograms.conf" ]
-            if "managesieve" in dovecot_protocols else [] }}' ]
+           --divert /etc/dovecot/dovecot.conf.dpkg-divert
+           --rename /etc/dovecot/dovecot.conf
+  args:
+    creates: /etc/dovecot/dovecot.conf.dpkg-divert
   notify: [ 'Restart dovecot' ]
+  tags: [ "role::dovecot:conf" ]
 
-- name: Remove redundant Dovecot configurations
-  file:
-    dest: '/etc/dovecot/conf.d/{{ item }}'
-    state: absent
-  with_flattened:
-    - [ '{{ "20-imap.conf" if "imap" not in dovecot_protocols else [] }}' ]
-    - [ '{{ "20-pop3.conf" if "pop3" not in dovecot_protocols else [] }}' ]
-    - [ '{{ "20-lmtp.conf" if "lmtp" not in dovecot_protocols else [] }}' ]
-    - [ '{{ [ "20-managesieve.conf", "90-sieve.conf", "90-sieve-extprograms.conf" ]
-            if "managesieve" not in dovecot_protocols else [] }}' ]
+- name: Generate Dovecot main configuration file
+  template:
+    src: '{{ lookup("template_src", "etc/dovecot/dovecot.conf.j2") }}'
+    dest: '/etc/dovecot/dovecot.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0644'
   notify: [ 'Restart dovecot' ]
+  tags: [ "role::dovecot:conf" ]
```

Comment:
This dovecot.conf templating task should probably be delayed until after the tasks for creating the 'vmail' user and group and other "dependent" tasks have been executed. It should probably be one of the last tasks. This way, when the playbook is interrupted, there are less errors for the administrator to deal with.

Reply:
Sure, I hope the new location is better (didn't know whether the `post_main` hook should go before or after the flushing of the event handlers).